STYLE_GUIDE.md

```diff
@@ -154,6 +154,62 @@ Prefer:
 
 ## JavaScript
 
+### Variable naming
+
+In order to allow Kedro-Viz to be worked on by many developers from different backgrounds, we try to follow a consistent set of naming conventions. Beyond consistency of form, element names must be easily understood and must convey the function of each element.
+
+✅ DO choose easily readable identifier names.
+
+For example, a variable named `horizontalAlignment` is more English-readable than `alignmentHorizontal`.
+
+✅ DO favour readability over brevity.
+
+The variable name `canScrollHorizontally` is better than `scrollableX`.
+
+✅ DO use verbs in the imperative mood for function/method names.
+
+The function `getSidebarLength()` is better than `sidebarLength()`, as it indicates how it should be used.
+
+✅ DO prefix Boolean variable names to indicate their type, and prefer positive names.
+
+This helps the reader easily infer that this variable is a Boolean value, and helps avoid complicated double-negatives.
+
+Avoid:
+
+```js
+let person = true;
+let age = true;
+let dance = true;
+let notDancing = true;
+```
+
+Prefer:
+
+```js
+let isPerson = true;
+let hasAge = true;
+let canDance = true;
+let isDancing = false;
+```
+
+✅ DO use camelCase for most variable names.
+
+✅ DO use PascalCase for class names.
+
+✅ DO use UPPER_CASE for constant values.
+
+✅ DO use camel-case for file names.
```

Comment:
this should be kebab-case? http://wiki.c2.com/?KebabCase

Reply:
God you're right, thanks good catch!